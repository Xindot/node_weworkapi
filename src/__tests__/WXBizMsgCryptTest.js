const jsSHA = require("jssha");
const WXBizMsgCrypt = require("../WXBizMsgCrypt");
jest.mock("../RandomGenerator");

describe("WXBizMsgCrypt TestSuite", () => {
    describe("verifyUrl TestCase", () => {
        it("Test WXBizMsgCrypt", () => {
            const corpId = "wx5823bf96d3bd56c7";
            const token = "QDG6eK";
            const encodingAesKey =
                "jWmYm7qr5nMoAUwZRjGtBxmz3KA1tkAj3ykkR6q2B2C";
            const { msg_signature, timestamp, nonce, echostr } = {
                msg_signature: "477715d11cdb4164915debcba66cb864d751f3e6",
                timestamp: "1409659813",
                nonce: "1372623149",
                echostr:
                    "RypEvHKD8QQKFhvQ6QleEB4J58tiPdvo+rtK1I9qca6aM/wvqnLSV5zEPeusUiX5L5X/0lWfrf0QADHHhGd3QczcdCUpj911L3vg3W/sYYvuJTs3TUUkSUXxaccAS0qhxchrRYt66wiSpGLYL42aM6A8dTT+6k4aSknmPj48kzJs8qLjvd4Xgpue06DOdnLxAUHzM6+kDZ+HMZfJYuR+LtwGc2hgf5gsijff0ekUNXZiqATP7PF5mZxZ3Izoun1s4zG4LUMnvw2r+KqCKIw+3IQH03v+BCA9nMELNqbSf6tiWSrXJB3LAVGUcallcrw8V2t9EL4EhzJWrQUax5wLVMNS0+rUPA3k22Ncx4XXZS9o0MBH27Bo6BpNelZpS+/uh9KsNlY6bHCmJU9p8g7m3fVKn28H3KDYA5Pl/T8Z1ptDAVe0lXdQ2YoyyH2uyPIGHBZZIs2pDBS8R07+qN+E7Q=="
            };
            let wxBizMsgCrypt = WXBizMsgCrypt(token, encodingAesKey, corpId);
            let content = wxBizMsgCrypt.verifyUrl(
                msg_signature,
                timestamp,
                nonce,
                echostr
            );
            expect(content).toMatchSnapshot();
        });
    });
    describe("decryptMessage TestCase", () => {
        it("Test decryptMessage", async () => {
            let corpId = "wx5823bf96d3bd56c7",
                token = "QDG6eK",
                encodingAesKey = "jWmYm7qr5nMoAUwZRjGtBxmz3KA1tkAj3ykkR6q2B2C",
                postData = `<xml>
            <ToUserName><![CDATA[wx5823bf96d3bd56c7]]></ToUserName><Encrypt><![CDATA[RypEvHKD8QQKFhvQ6QleEB4J58tiPdvo+rtK1I9qca6aM/wvqnLSV5zEPeusUiX5L5X/0lWfrf0QADHHhGd3QczcdCUpj911L3vg3W/sYYvuJTs3TUUkSUXxaccAS0qhxchrRYt66wiSpGLYL42aM6A8dTT+6k4aSknmPj48kzJs8qLjvd4Xgpue06DOdnLxAUHzM6+kDZ+HMZfJYuR+LtwGc2hgf5gsijff0ekUNXZiqATP7PF5mZxZ3Izoun1s4zG4LUMnvw2r+KqCKIw+3IQH03v+BCA9nMELNqbSf6tiWSrXJB3LAVGUcallcrw8V2t9EL4EhzJWrQUax5wLVMNS0+rUPA3k22Ncx4XXZS9o0MBH27Bo6BpNelZpS+/uh9KsNlY6bHCmJU9p8g7m3fVKn28H3KDYA5Pl/T8Z1ptDAVe0lXdQ2YoyyH2uyPIGHBZZIs2pDBS8R07+qN+E7Q==]]></Encrypt>
            <AgentID><![CDATA[218]]></AgentID>
            </xml>`,
                signature = "477715d11cdb4164915debcba66cb864d751f3e6",
                timestamps = "1409659813",
                nonce = "1372623149";
            let wxBizMsgCrypt = WXBizMsgCrypt(token, encodingAesKey, corpId);
            let message = await wxBizMsgCrypt.decryptMessage(
                signature,
                timestamps,
                nonce,
                postData
            );
            expect(message).toMatchSnapshot();
        });
    });
    describe("encryptMessage TestCase", () => {
        it("Test encryptMessage", () => {
            let corpId = "ww1436e0e65a779aee";
            let token = "hJqcu3uJ9Tn2gXPmxx2w9kkCkCE2EPYo";
            let encodingAesKey = "6qkdMrq68nTKduznJYO1A37W2oEgpkMUvkttRToqhUt";
            let signature = "0c3914025cb4b4d68103f6bfc8db550f79dcf48e";
            let timestamp = "1476422779";
            let nonce = "1597212914";
            let replyMsg =
                "<xml><ToUserName>ww1436e0e65a779aee</ToUserName><FromUserName>ChenJiaShun</FromUserName><CreateTime>1476422779</CreateTime><MsgType>text</MsgType><Content>你好</Content><MsgId>1456453720</MsgId><AgentID>1000002</AgentID></xml>";
            debugger;
            let wxBizMsgCrypt = WXBizMsgCrypt(token, encodingAesKey, corpId);
            let message = wxBizMsgCrypt.encryptMessage(
                replyMsg,
                timestamp,
                nonce
            );
            expect(message).toMatchSnapshot();
        });
    });
});
