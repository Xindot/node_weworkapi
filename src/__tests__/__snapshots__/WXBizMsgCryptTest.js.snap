// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WXBizMsgCrypt TestSuite decryptMessage TestCase Test decryptMessage 1`] = `
"<xml><ToUserName><![CDATA[wx5823bf96d3bd56c7]]></ToUserName>
<FromUserName><![CDATA[mycreate]]></FromUserName>
<CreateTime>1409659813</CreateTime>
<MsgType><![CDATA[text]]></MsgType>
<Content><![CDATA[hello]]></Content>
<MsgId>4561255354251345929</MsgId>
<AgentID>218</AgentID>
</xml>"
`;

exports[`WXBizMsgCrypt TestSuite encryptMessage TestCase Test encryptMessage 1`] = `
"<xml>
<Encrypt><![CDATA[RS76Qh1DmrbNRFfnby47FdyStlVxXaGGc7vQscUqWnEo2oN65g8FJaQsNqEsmyC0gd2+m56P0XeYZTqWLSuVKihnJ4kbaxOkV5XYbCAGx//HvRR5k0J4SPHHiKBn3y2OuEjQOXKbuu+a8I/P/2Pm8rMJoPSyZMotXcox3GFUtyW+ZOVwU/v2MEAeu3iWFm4Gd4CacPVRa21E2G18jnMJnMroiXrZcF1rgsXjiHnb+umIti7TTtRn4/S61g9SbhWMyruDaW/hhfdDAT2PXgcE7J+cZ/x4fw0GHfpTCfygYF/92JVxLULGUIcEa349UD+req+6k7sE9rimfOAeLQglRlj6smUs6ADRX/b/9VNG9tFcMIIb7uw2cMyDGs4hQD+J]]></Encrypt>
<MsgSignature><![CDATA[daadc8ba55ba31446ab8d9a4f6f0f8b491f92249]]></MsgSignature>
<TimeStamp>1476422779</TimeStamp>
<Nonce><![CDATA[1597212914]]></Nonce>
</xml>"
`;

exports[`WXBizMsgCrypt TestSuite verifyUrl TestCase Test WXBizMsgCrypt 1`] = `
"<xml><ToUserName><![CDATA[wx5823bf96d3bd56c7]]></ToUserName>
<FromUserName><![CDATA[mycreate]]></FromUserName>
<CreateTime>1409659813</CreateTime>
<MsgType><![CDATA[text]]></MsgType>
<Content><![CDATA[hello]]></Content>
<MsgId>4561255354251345929</MsgId>
<AgentID>218</AgentID>
</xml>"
`;
